src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java

```diff
@@ -328,6 +362,29 @@ else if (ast.getType() == TokenTypes.METHOD_DEF) {
         return result;
     }
 
+    /**
+     * Checks if a variable is to be ignored based on its modifiers.
+     * @param modifiers modifiers of the variable to be checked
+     * @return true if there is a modifier to be ignored
+     */
+    private boolean hasIgnoredModifiers(DetailAST modifiers) {
+        final boolean isStatic = modifiers.findFirstToken(TokenTypes.LITERAL_STATIC) != null;
+        final boolean isFinal = modifiers.findFirstToken(TokenTypes.FINAL) != null;
+        final boolean ignoredStatic = ignoreStatic && isStatic;
+        final boolean ignoredFinal = ignoreFinal && isFinal;
+        final boolean result;
+        if (ignoreStaticFinal) {
+            result = isStatic && isFinal
+                || ignoredFinal
+                || ignoredStatic;
+        }
+        else {
+            result = ignoredFinal && !isStatic
+                || ignoredStatic && !isFinal;
```

Comment:
please simplify boolean expressions above. Here `ignoredStatic` and `ignoredFinal` contain `isStatic` and `isFinal` in their initialization.

Reply:
I rewrote the boolean operations. It should be easier to understand now.